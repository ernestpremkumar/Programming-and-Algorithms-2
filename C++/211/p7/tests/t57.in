print
insert 6
print
balanced
insert 6
size
balanced
size
echo 3
balanced
print
insert 3
print
insert 1
insert 2
print
balanced
insert 5
breadth
insert 0
echo 6
distance
print
size
breadth
distance
print
echo 7
echo 1
find 2
find 1
distance
breadth
breadth
echo 2
insert 6
size
balanced
echo 6
find 1
size
balanced
echo 7
echo 7
insert 0
size
find 4
distance
echo 0
size
size
echo 3
find 4
find 7
echo 3
insert 6
balanced
insert 0
balanced
balanced
breadth
breadth
size
echo 4
balanced
size
distance
distance
insert 1
print
balanced
find 5
breadth
breadth
distance
distance
distance
find 3
balanced
size
distance
echo 4
distance
breadth
balanced
echo 1
balanced
print
echo 4
insert 1
distance
size
find 4
print
insert 2
insert 6
size
find 2
echo 0
balanced
size
distance
size
breadth
print
size
distance
distance
size
print
echo 4
find 5
balanced
size
balanced
echo 1
echo 1
size
insert 3
distance
balanced
echo 0
print
balanced
insert 7
find 3
echo 2
print
breadth
echo 7
echo 5
distance
distance
print
insert 0
distance
size
breadth
breadth
breadth
balanced
insert 0
balanced
echo 5
print
insert 0
echo 5
distance
insert 6
distance
print
size
find 6
print
print
balanced
size
find 0
distance
distance
find 2
balanced
balanced
size
balanced
size
breadth
insert 4
print
echo 6
echo 0
find 0
echo 2
print
size
insert 2
insert 6
insert 0
distance
insert 3
print
insert 4
echo 1
insert 2
echo 4
print
insert 0
find 0
balanced
echo 7
insert 1
balanced
size
find 1
print
breadth
insert 0
insert 6
insert 1
distance
print
size
size
balanced
print
find 0
insert 5
balanced
find 3
find 3
echo 0
echo 1
balanced
size
print
size
find 0
distance
insert 5
balanced
breadth
size
print
size
print
echo 6
balanced
breadth
size
balanced
echo 7
balanced
breadth
insert 5
find 6
find 0
size
balanced
distance
distance
echo 3
balanced
balanced
breadth
breadth
echo 4
breadth
insert 0
breadth
echo 2
size
insert 3
echo 4
size
size
size
balanced
insert 1
distance
echo 7
find 3
insert 0
echo 1
find 2
print
distance
balanced
size
echo 0
balanced
breadth
breadth
size
find 6
breadth
print
print
size
balanced
size
insert 3
insert 1
echo 1
insert 3
echo 0
find 0
insert 5
size
distance
print
insert 4
breadth
distance
size
print
insert 4
breadth
balanced
distance
distance
echo 6
print
insert 1
print
print
print
size
find 1
echo 0
distance
size
size
find 7
breadth
insert 1
echo 4
breadth
size
size
balanced
echo 7
breadth
print
size
distance
echo 1
insert 0
balanced
print
breadth
insert 6
balanced
insert 6
breadth
insert 4
find 4
balanced
find 0
echo 3
balanced
distance
distance
find 7
find 6
find 7
breadth
distance
size
balanced
print
print
insert 7
balanced
balanced
insert 7
distance
balanced
balanced
size
insert 3
print
insert 7
print
size
print
find 6
breadth
print
echo 4
print
find 1
insert 2
distance
print
print
distance
print
size
print
find 5
find 0
echo 6
size
breadth
insert 2
print
find 4
distance
distance
find 2
balanced
size
insert 4
print
size
breadth
distance
distance
find 1
find 4
breadth
echo 7
breadth
find 6
echo 3
distance
balanced
find 0
breadth
distance
echo 7
breadth
size
balanced
balanced
insert 3
size
distance
balanced
find 1
balanced
distance
insert 6
balanced
size
distance
size
echo 3
balanced
insert 0
find 6
find 2
find 5
breadth
size
insert 0
insert 1
echo 7
insert 2
insert 7
breadth
breadth
find 7
find 6
distance
insert 4
distance
distance
distance
find 3
find 0
size
print
balanced
insert 5
insert 0
echo 7
distance
print
distance
distance
find 7
print
distance
breadth
size
distance
breadth
distance
echo 4
breadth
balanced
balanced
insert 5
balanced
distance
balanced
find 2
print
breadth
find 1
size
print
print
print
find 4
breadth
balanced
balanced
insert 0
echo 7
breadth
insert 0
find 0
insert 5
balanced
insert 5
balanced
insert 5
echo 3
echo 6
balanced
breadth
size
balanced
print
find 7
balanced
insert 4
size
print
balanced
size
size
size
size
distance
balanced
echo 4
insert 2
breadth
distance
distance
breadth
insert 7
insert 7
echo 5
size
balanced
distance
balanced
find 0
size
size
breadth
insert 3
echo 2
find 7
breadth
breadth
balanced
find 6
balanced
insert 5
size
echo 0
echo 1
balanced
balanced
print
print
distance
breadth
size
insert 2
insert 7
distance
balanced
size
print
insert 4
breadth
insert 1
print
distance
print
distance
balanced
print
breadth
insert 5
distance
find 7
breadth
distance
insert 1
distance
insert 6
find 7
balanced
breadth
breadth
distance
distance
distance
balanced
size
balanced
find 5
size
distance
insert 3
size
print
balanced
balanced
balanced
echo 4
insert 5
balanced
size
balanced
print
size
find 7
echo 5
echo 1
distance
find 2
echo 2
breadth
balanced
print
insert 7
breadth
print
balanced
breadth
find 3
size
print
echo 2
print
balanced
echo 7
size
balanced
size
echo 3
echo 3
print
breadth
echo 3
distance
size
size
size
find 2
echo 6
distance
size
breadth
insert 1
print
find 1
distance
insert 7
size
distance
breadth
echo 0
balanced
insert 2
insert 2
breadth
insert 7
balanced
insert 6
distance
insert 6
size
echo 7
find 3
breadth
size
size
insert 2
print
print
find 5
print
insert 4
size
find 3
insert 0
insert 4
print
balanced
find 7
insert 4
distance
balanced
breadth
insert 4
insert 1
find 4
echo 6
insert 4
balanced
find 5
insert 0
print
distance
breadth
balanced
balanced
size
print
size
insert 5
insert 4
size
insert 1
find 3
echo 0
insert 6
find 2
balanced
echo 1
size
find 7
print
echo 5
breadth
breadth
distance
find 6
balanced
print
breadth
insert 4
distance
find 4
breadth
size
size
insert 5
find 6
distance
find 0
balanced
insert 2
size
balanced
size
find 4
size
balanced
balanced
echo 4
print
balanced
echo 5
distance
find 6
insert 1
find 2
breadth
size
print
insert 5
echo 7
insert 4
insert 0
print
size
print
breadth
print
find 3
breadth
size
balanced
distance
print
balanced
size
size
balanced
print
breadth
print
echo 2
balanced
breadth
print
balanced
find 6
distance
insert 1
print
echo 0
balanced
size
distance
find 6
breadth
breadth
insert 5
size
balanced
insert 7
breadth
breadth
distance
distance
distance
insert 6
insert 6
breadth
size
size
balanced
breadth
print
echo 7
distance
size
distance
balanced
distance
distance
print
size
print
print
balanced
distance
distance
echo 3
size
distance
insert 1
breadth
echo 7
find 4
size
size
size
find 5
print
print
breadth
find 2
breadth
balanced
find 0
insert 5
distance
print
size
print
insert 4
size
find 3
insert 5
print
print
echo 2
distance
size
distance
size
insert 7
breadth
distance
breadth
print
insert 6
size
print
print
insert 0
size
balanced
echo 0
breadth
distance
breadth
echo 0
breadth
insert 7
insert 2
insert 5
print
balanced
distance
echo 3
print
size
find 3
breadth
echo 1
insert 3
find 5
balanced
echo 1
find 0
distance
print
balanced
size
balanced
distance
breadth
distance
size
size
balanced
balanced
print
find 6
distance
balanced
find 4
distance
balanced
insert 0
find 6
insert 7
balanced
insert 7
find 1
find 3
distance
print
echo 3
echo 0
balanced
find 7
size
breadth
size
balanced
breadth
breadth
echo 4
breadth
size
breadth
size
print
insert 2
insert 4
find 0
find 4
breadth
balanced
balanced
print
balanced
insert 2
insert 7
size
balanced
print
echo 1
print
print
size
insert 7
breadth
distance
distance
size
insert 2
print
echo 2
insert 5
insert 2
insert 0
insert 5
distance
print
size
size
print
size
find 2
balanced
size
balanced
find 7
print
size
print
find 3
breadth
find 1
find 1
echo 5
insert 5
print
distance
find 7
breadth
breadth
print
find 0
print
distance
insert 0
distance
size
echo 1
distance
find 0
print
size
balanced
