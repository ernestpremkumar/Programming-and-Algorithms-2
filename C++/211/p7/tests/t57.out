{}
{6}
Tree is balanced.
1
Tree is balanced.
1
3
Tree is balanced.
{6}
{3, 6}
{1, 2, 3, 6}
Tree is not balanced.
{6, 3, 1, 5, 2}
6
Average distance of nodes to root = 1.83333
{0, 1, 2, 3, 5, 6}
6
{6, 3, 1, 5, 0, 2}
Average distance of nodes to root = 1.83333
{0, 1, 2, 3, 5, 6}
7
1
<2> is in tree.
<1> is in tree.
Average distance of nodes to root = 1.83333
{6, 3, 1, 5, 0, 2}
{6, 3, 1, 5, 0, 2}
2
6
Tree is not balanced.
6
<1> is in tree.
6
Tree is not balanced.
7
7
6
<4> is not in tree.
Average distance of nodes to root = 1.83333
0
6
6
3
<4> is not in tree.
<7> is not in tree.
3
Tree is not balanced.
Tree is not balanced.
Tree is not balanced.
{6, 3, 1, 5, 0, 2}
{6, 3, 1, 5, 0, 2}
6
4
Tree is not balanced.
6
Average distance of nodes to root = 1.83333
Average distance of nodes to root = 1.83333
{0, 1, 2, 3, 5, 6}
Tree is not balanced.
<5> is in tree.
{6, 3, 1, 5, 0, 2}
{6, 3, 1, 5, 0, 2}
Average distance of nodes to root = 1.83333
Average distance of nodes to root = 1.83333
Average distance of nodes to root = 1.83333
<3> is in tree.
Tree is not balanced.
6
Average distance of nodes to root = 1.83333
4
Average distance of nodes to root = 1.83333
{6, 3, 1, 5, 0, 2}
Tree is not balanced.
1
Tree is not balanced.
{0, 1, 2, 3, 5, 6}
4
Average distance of nodes to root = 1.83333
6
<4> is not in tree.
{0, 1, 2, 3, 5, 6}
6
<2> is in tree.
0
Tree is not balanced.
6
Average distance of nodes to root = 1.83333
6
{6, 3, 1, 5, 0, 2}
{0, 1, 2, 3, 5, 6}
6
Average distance of nodes to root = 1.83333
Average distance of nodes to root = 1.83333
6
{0, 1, 2, 3, 5, 6}
4
<5> is in tree.
Tree is not balanced.
6
Tree is not balanced.
1
1
6
Average distance of nodes to root = 1.83333
Tree is not balanced.
0
{0, 1, 2, 3, 5, 6}
Tree is not balanced.
<3> is in tree.
2
{0, 1, 2, 3, 5, 6, 7}
{6, 3, 7, 1, 5, 0, 2}
7
5
Average distance of nodes to root = 1.71429
Average distance of nodes to root = 1.71429
{0, 1, 2, 3, 5, 6, 7}
Average distance of nodes to root = 1.71429
7
{6, 3, 7, 1, 5, 0, 2}
{6, 3, 7, 1, 5, 0, 2}
{6, 3, 7, 1, 5, 0, 2}
Tree is not balanced.
Tree is not balanced.
5
{0, 1, 2, 3, 5, 6, 7}
5
Average distance of nodes to root = 1.71429
Average distance of nodes to root = 1.71429
{0, 1, 2, 3, 5, 6, 7}
7
<6> is in tree.
{0, 1, 2, 3, 5, 6, 7}
{0, 1, 2, 3, 5, 6, 7}
Tree is not balanced.
7
<0> is in tree.
Average distance of nodes to root = 1.71429
Average distance of nodes to root = 1.71429
<2> is in tree.
Tree is not balanced.
Tree is not balanced.
7
Tree is not balanced.
7
{6, 3, 7, 1, 5, 0, 2}
{0, 1, 2, 3, 4, 5, 6, 7}
6
0
<0> is in tree.
2
{0, 1, 2, 3, 4, 5, 6, 7}
8
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
1
4
{0, 1, 2, 3, 4, 5, 6, 7}
<0> is in tree.
Tree is not balanced.
7
Tree is not balanced.
8
<1> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 3, 7, 1, 5, 0, 2, 4}
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
<0> is in tree.
Tree is not balanced.
<3> is in tree.
<3> is in tree.
0
1
Tree is not balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
<0> is in tree.
Average distance of nodes to root = 1.875
Tree is not balanced.
{6, 3, 7, 1, 5, 0, 2, 4}
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
6
Tree is not balanced.
{6, 3, 7, 1, 5, 0, 2, 4}
8
Tree is not balanced.
7
Tree is not balanced.
{6, 3, 7, 1, 5, 0, 2, 4}
<6> is in tree.
<0> is in tree.
8
Tree is not balanced.
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
3
Tree is not balanced.
Tree is not balanced.
{6, 3, 7, 1, 5, 0, 2, 4}
{6, 3, 7, 1, 5, 0, 2, 4}
4
{6, 3, 7, 1, 5, 0, 2, 4}
{6, 3, 7, 1, 5, 0, 2, 4}
2
8
4
8
8
8
Tree is not balanced.
Average distance of nodes to root = 1.875
7
<3> is in tree.
1
<2> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
Tree is not balanced.
8
0
Tree is not balanced.
{6, 3, 7, 1, 5, 0, 2, 4}
{6, 3, 7, 1, 5, 0, 2, 4}
8
<6> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
8
Tree is not balanced.
8
1
0
<0> is in tree.
8
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 3, 7, 1, 5, 0, 2, 4}
Average distance of nodes to root = 1.875
8
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 3, 7, 1, 5, 0, 2, 4}
Tree is not balanced.
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
6
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
8
<1> is in tree.
0
Average distance of nodes to root = 1.875
8
8
<7> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
4
{6, 3, 7, 1, 5, 0, 2, 4}
8
8
Tree is not balanced.
7
{6, 3, 7, 1, 5, 0, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
8
Average distance of nodes to root = 1.875
1
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 3, 7, 1, 5, 0, 2, 4}
Tree is not balanced.
{6, 3, 7, 1, 5, 0, 2, 4}
<4> is in tree.
Tree is not balanced.
<0> is in tree.
3
Tree is not balanced.
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
<7> is in tree.
<6> is in tree.
<7> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
Average distance of nodes to root = 1.875
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 1.875
Tree is not balanced.
Tree is not balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
<6> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
4
{0, 1, 2, 3, 4, 5, 6, 7}
<1> is in tree.
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
<5> is in tree.
<0> is in tree.
6
8
{6, 3, 7, 1, 5, 0, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
<4> is in tree.
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
<2> is in tree.
Tree is not balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
{6, 3, 7, 1, 5, 0, 2, 4}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
<1> is in tree.
<4> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
7
{6, 3, 7, 1, 5, 0, 2, 4}
<6> is in tree.
3
Average distance of nodes to root = 1.875
Tree is not balanced.
<0> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
Average distance of nodes to root = 1.875
7
{6, 3, 7, 1, 5, 0, 2, 4}
8
Tree is not balanced.
Tree is not balanced.
8
Average distance of nodes to root = 1.875
Tree is not balanced.
<1> is in tree.
Tree is not balanced.
Average distance of nodes to root = 1.875
Tree is not balanced.
8
Average distance of nodes to root = 1.875
8
3
Tree is not balanced.
<6> is in tree.
<2> is in tree.
<5> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
8
7
{6, 3, 7, 1, 5, 0, 2, 4}
{6, 3, 7, 1, 5, 0, 2, 4}
<7> is in tree.
<6> is in tree.
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
<3> is in tree.
<0> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
7
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
<7> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
{6, 3, 7, 1, 5, 0, 2, 4}
8
Average distance of nodes to root = 1.875
{6, 3, 7, 1, 5, 0, 2, 4}
Average distance of nodes to root = 1.875
4
{6, 3, 7, 1, 5, 0, 2, 4}
Tree is not balanced.
Tree is not balanced.
Tree is not balanced.
Average distance of nodes to root = 1.875
Tree is not balanced.
<2> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 3, 7, 1, 5, 0, 2, 4}
<1> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
<4> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
Tree is not balanced.
Tree is not balanced.
7
{6, 3, 7, 1, 5, 0, 2, 4}
<0> is in tree.
Tree is not balanced.
Tree is not balanced.
3
6
Tree is not balanced.
{6, 3, 7, 1, 5, 0, 2, 4}
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
<7> is in tree.
Tree is not balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
8
8
8
8
Average distance of nodes to root = 1.875
Tree is not balanced.
4
{6, 3, 7, 1, 5, 0, 2, 4}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
{6, 3, 7, 1, 5, 0, 2, 4}
5
8
Tree is not balanced.
Average distance of nodes to root = 1.875
Tree is not balanced.
<0> is in tree.
8
8
{6, 3, 7, 1, 5, 0, 2, 4}
2
<7> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
{6, 3, 7, 1, 5, 0, 2, 4}
Tree is not balanced.
<6> is in tree.
Tree is not balanced.
8
0
1
Tree is not balanced.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
{6, 3, 7, 1, 5, 0, 2, 4}
8
Average distance of nodes to root = 1.875
Tree is not balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 3, 7, 1, 5, 0, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 3, 7, 1, 5, 0, 2, 4}
Average distance of nodes to root = 1.875
<7> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
<7> is in tree.
Tree is not balanced.
{6, 3, 7, 1, 5, 0, 2, 4}
{6, 3, 7, 1, 5, 0, 2, 4}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
Tree is not balanced.
8
Tree is not balanced.
<5> is in tree.
8
Average distance of nodes to root = 1.875
8
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Tree is not balanced.
Tree is not balanced.
4
Tree is not balanced.
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
8
<7> is in tree.
5
1
Average distance of nodes to root = 1.875
<2> is in tree.
2
{6, 3, 7, 1, 5, 0, 2, 4}
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 3, 7, 1, 5, 0, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
{6, 3, 7, 1, 5, 0, 2, 4}
<3> is in tree.
8
{0, 1, 2, 3, 4, 5, 6, 7}
2
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
7
8
Tree is not balanced.
8
3
3
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 3, 7, 1, 5, 0, 2, 4}
3
Average distance of nodes to root = 1.875
8
8
8
<2> is in tree.
6
Average distance of nodes to root = 1.875
8
{6, 3, 7, 1, 5, 0, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
<1> is in tree.
Average distance of nodes to root = 1.875
8
Average distance of nodes to root = 1.875
{6, 3, 7, 1, 5, 0, 2, 4}
0
Tree is not balanced.
{6, 3, 7, 1, 5, 0, 2, 4}
Tree is not balanced.
Average distance of nodes to root = 1.875
8
7
<3> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
8
8
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
<5> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
8
<3> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
<7> is in tree.
Average distance of nodes to root = 1.875
Tree is not balanced.
{6, 3, 7, 1, 5, 0, 2, 4}
<4> is in tree.
6
Tree is not balanced.
<5> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
{6, 3, 7, 1, 5, 0, 2, 4}
Tree is not balanced.
Tree is not balanced.
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
<3> is in tree.
0
<2> is in tree.
Tree is not balanced.
1
8
<7> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
5
{6, 3, 7, 1, 5, 0, 2, 4}
{6, 3, 7, 1, 5, 0, 2, 4}
Average distance of nodes to root = 1.875
<6> is in tree.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 3, 7, 1, 5, 0, 2, 4}
Average distance of nodes to root = 1.875
<4> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
8
8
<6> is in tree.
Average distance of nodes to root = 1.875
<0> is in tree.
Tree is not balanced.
8
Tree is not balanced.
8
<4> is in tree.
8
Tree is not balanced.
Tree is not balanced.
4
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
5
Average distance of nodes to root = 1.875
<6> is in tree.
<2> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
8
{0, 1, 2, 3, 4, 5, 6, 7}
7
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 3, 7, 1, 5, 0, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
<3> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
8
Tree is not balanced.
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
8
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 3, 7, 1, 5, 0, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
2
Tree is not balanced.
{6, 3, 7, 1, 5, 0, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
<6> is in tree.
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
0
Tree is not balanced.
8
Average distance of nodes to root = 1.875
<6> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
{6, 3, 7, 1, 5, 0, 2, 4}
8
Tree is not balanced.
{6, 3, 7, 1, 5, 0, 2, 4}
{6, 3, 7, 1, 5, 0, 2, 4}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
{6, 3, 7, 1, 5, 0, 2, 4}
8
8
Tree is not balanced.
{6, 3, 7, 1, 5, 0, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
7
Average distance of nodes to root = 1.875
8
Average distance of nodes to root = 1.875
Tree is not balanced.
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
3
8
Average distance of nodes to root = 1.875
{6, 3, 7, 1, 5, 0, 2, 4}
7
<4> is in tree.
8
8
8
<5> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
{6, 3, 7, 1, 5, 0, 2, 4}
<2> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
Tree is not balanced.
<0> is in tree.
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
<3> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
2
Average distance of nodes to root = 1.875
8
Average distance of nodes to root = 1.875
8
{6, 3, 7, 1, 5, 0, 2, 4}
Average distance of nodes to root = 1.875
{6, 3, 7, 1, 5, 0, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
8
Tree is not balanced.
0
{6, 3, 7, 1, 5, 0, 2, 4}
Average distance of nodes to root = 1.875
{6, 3, 7, 1, 5, 0, 2, 4}
0
{6, 3, 7, 1, 5, 0, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
Average distance of nodes to root = 1.875
3
{0, 1, 2, 3, 4, 5, 6, 7}
8
<3> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
1
<5> is in tree.
Tree is not balanced.
1
<0> is in tree.
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
8
Tree is not balanced.
Average distance of nodes to root = 1.875
{6, 3, 7, 1, 5, 0, 2, 4}
Average distance of nodes to root = 1.875
8
8
Tree is not balanced.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
<6> is in tree.
Average distance of nodes to root = 1.875
Tree is not balanced.
<4> is in tree.
Average distance of nodes to root = 1.875
Tree is not balanced.
<6> is in tree.
Tree is not balanced.
<1> is in tree.
<3> is in tree.
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
3
0
Tree is not balanced.
<7> is in tree.
8
{6, 3, 7, 1, 5, 0, 2, 4}
8
Tree is not balanced.
{6, 3, 7, 1, 5, 0, 2, 4}
{6, 3, 7, 1, 5, 0, 2, 4}
4
{6, 3, 7, 1, 5, 0, 2, 4}
8
{6, 3, 7, 1, 5, 0, 2, 4}
8
{0, 1, 2, 3, 4, 5, 6, 7}
<0> is in tree.
<4> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
Tree is not balanced.
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
Tree is not balanced.
8
Tree is not balanced.
{0, 1, 2, 3, 4, 5, 6, 7}
1
{0, 1, 2, 3, 4, 5, 6, 7}
{0, 1, 2, 3, 4, 5, 6, 7}
8
{6, 3, 7, 1, 5, 0, 2, 4}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
8
{0, 1, 2, 3, 4, 5, 6, 7}
2
Average distance of nodes to root = 1.875
{0, 1, 2, 3, 4, 5, 6, 7}
8
8
{0, 1, 2, 3, 4, 5, 6, 7}
8
<2> is in tree.
Tree is not balanced.
8
Tree is not balanced.
<7> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
8
{0, 1, 2, 3, 4, 5, 6, 7}
<3> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
<1> is in tree.
<1> is in tree.
5
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
<7> is in tree.
{6, 3, 7, 1, 5, 0, 2, 4}
{6, 3, 7, 1, 5, 0, 2, 4}
{0, 1, 2, 3, 4, 5, 6, 7}
<0> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
Average distance of nodes to root = 1.875
Average distance of nodes to root = 1.875
8
1
Average distance of nodes to root = 1.875
<0> is in tree.
{0, 1, 2, 3, 4, 5, 6, 7}
8
Tree is not balanced.
